tiledb/sm/c_api/tiledb.cc

```diff
@@ -1437,29 +1437,16 @@ int32_t tiledb_array_get_open_timestamp_end(
 
 int32_t tiledb_array_delete(tiledb_ctx_t* ctx, const char* uri) {
   // Allocate an array object
-  tiledb_array_t* array = new (std::nothrow) tiledb_array_t;
-  try {
-    array->array_ = make_shared<tiledb::sm::Array>(
-        HERE(), ctx->resources(), tiledb::sm::URI(uri));
-  } catch (std::bad_alloc&) {
-    auto st = Status_Error(
-        "Failed to create TileDB array object; Memory allocation error");
-    delete array;
-    array = nullptr;
-    LOG_STATUS_NO_RETURN_VALUE(st);
-    save_error(ctx, st);
-    return TILEDB_OOM;
-  }
-
+  tiledb::sm::Array array(ctx->resources(), tiledb::sm::URI(uri));
   // Open the array for exclusive modification
-  throw_if_not_ok(array->array_->open(
+  throw_if_not_ok(array.open(
       tiledb::sm::QueryType::MODIFY_EXCLUSIVE,
       tiledb::sm::EncryptionType::NO_ENCRYPTION,
       nullptr,
       0));
 
   try {
```

Comment:
As long as you're modifying the code, we can eliminate this `try` statement, which no longer does anything that the top-level exception handler doesn't already do.